app/src/main/java/com/rkm/tasky/util/storage/preferences/implementation/UserPreferencesRepositoryImpl.kt

```diff
@@ -0,0 +1,39 @@
+package com.rkm.tasky.util.storage.preferences.implementation
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import com.rkm.tasky.util.storage.preferences.abstraction.UserPreferencesRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+class UserPreferencesRepositoryImpl @Inject constructor(
+   private val dataStore: DataStore<Preferences>
+): UserPreferencesRepository {
+
+    private val keyMap = mapOf(
+        USER_KEY_ACCESS_TOKEN to stringPreferencesKey(USER_KEY_ACCESS_TOKEN),
+        USER_KEY_RESPONSE_TOKEN to stringPreferencesKey(USER_KEY_RESPONSE_TOKEN)
+    )
+
+    override suspend fun saveValue(key: String, value: String) {
+        dataStore.edit { preferences ->
+           preferences[keyMap[key]!!] = value
```

Comment:
You will get a crash here when a value for `key` other than your two constants is received. This can easily happen, becuase the type for `key` is String and not a type like an `Enum` or `sealed class` with a fixed set of values